x/reward/abci.go

```diff
@@ -1,37 +1,45 @@
 package reward
 
 import (
-	abci "github.com/cometbft/cometbft/abci/types"
+	"context"
+
+	sdkmath "cosmossdk.io/math"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
 	disttypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+
 	"github.com/xpladev/xpla/x/reward/keeper"
 	"github.com/xpladev/xpla/x/reward/types"
-
-	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
-	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
 )
 
-func BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock, k keeper.Keeper, bk types.BankKeeper, sk types.StakingKeeper, dk types.DistributionKeeper) {
+func BeginBlocker(ctx context.Context, k keeper.Keeper, bk types.BankKeeper, sk types.StakingKeeper, dk types.DistributionKeeper) error {
 	params := k.GetParams(ctx)
 
 	if params.RewardDistributeAccount == "" {
-		return
+		return nil
 	}
 
+	var err error = nil
+
 	total := params.TotalRate()
 
 	rewardDistributeAccount := sdk.MustAccAddressFromBech32(params.RewardDistributeAccount)
 
 	totalRewards := map[string]sdk.Coin{}
 
 	sk.IterateDelegations(ctx, rewardDistributeAccount, func(index int64, delegation stakingtypes.DelegationI) (stop bool) {
-		validator := delegation.GetValidatorAddr()
+		valAddr, err := sk.ValidatorAddressCodec().StringToBytes(delegation.GetValidatorAddr())
+		if err != nil {
+			return true
+		}
 
-		reward, err := dk.WithdrawDelegationRewards(ctx, rewardDistributeAccount, validator)
+		reward, err := dk.WithdrawDelegationRewards(ctx, rewardDistributeAccount, sdk.ValAddress(valAddr))
 		if err == disttypes.ErrEmptyDelegationDistInfo {
 			return false
 		} else if err != nil {
-			panic(err)
+			return true
```

Comment:
It would be better to write a log, since the chain continues to progress without halt with an error.

It's same to here
https://github.com/xpladev/xpla/pull/154/files#diff-ad6f2e18775e6aa226ba3ba4f7824ea403993769df34ef29bbf09facbbc5304bR35

Reply:
https://github.com/xpladev/xpla/pull/154/commits/2d646883b34f6c55defc2a1ec0402f37bc52b650